### PR TITLE
Finish OEP-18 compliance, and make work with latest DRF

### PR DESCRIPTION
**Pull request by nedbat**

### CODE REVIEW

**Comment by nedbat** [comment]
@jmbowman @feanil what's our best thinking about how to test a library like this? Spinning out tox envs for DRF and Django gets unwieldy.

File: test_settings.py [hunk 1]
```diff
@@ -36,5 +36,6 @@
 # DRF Settings
 REST_FRAMEWORK = {
     'PAGE_SIZE': 20,
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
```
Comment (nedbat) [inline test_settings.py]: @doctoryes turns out this was the thing breaking the test.  DRF 3.7 removed the default pagination class (and clearly labeled the change in the release notes...).

Comment (doctoryes) [reply test_settings.py]: Good find! And good to know. I feel like DRF could have deprecated/removed it in a more informative way - but fine.

**Comment by jmbowman** [comment]
@nedbat I don't see much point in testing multiple DRF versions here unless we expect it to be used with 2+ different DRF versions for a while to come; if we're just in the process of upgrading to a new DRF version, probably best to test against that and then go up to the repos that use this and do the same there.

**Comment by feanil** [comment]
@jmbowman part of the reason we're testing multiple versions for now is that going up to the latest from where we are is risky as DRF is so big and critical.  My initial goal was to get us up to the next version while making it easier to go up further with more confidence.

**Comment by feanil** [comment]
Gonna merge this to un-block other work for now though if we feel strongly about not matrixing in DRF(even if it is just until we get to the latest version) then I'm fine un-doing that change later.  The tests are fast on this repo so not sure how much of a cost there is on being thorough here.

**Pull request merged**